compiler/src/dotty/tools/dotc/transform/localopt/StringInterpolatorOpt.scala

```diff
@@ -74,14 +74,24 @@ class StringInterpolatorOpt extends MiniPhase {
         case SOrRawInterpolator(strs, elems) =>
           if (tree.symbol == defn.StringContext_raw) Some(strs, elems)
           else { // tree.symbol == defn.StringContextS
+            import dotty.tools.dotc.util.SourcePosition
+            var stringPosition: SourcePosition = null
             try {
               val escapedStrs = strs.map { str =>
-                val escapedValue = StringContext.processEscapes(str.const.stringValue)
-                cpy.Literal(str)(Constant(escapedValue))
+                stringPosition = str.sourcePos // TODO: + length of quote(s)? Why?
+                val escaped = StringContext.processEscapes(str.const.stringValue)
+                cpy.Literal(str)(Constant(escaped))
               }
               Some(escapedStrs, elems)
             } catch {
-              case _: StringContext.InvalidEscapeException => None
+              case iee: StringContext.InvalidEscapeException => {
+                ctx.error(iee.getMessage() + "\n", stringPosition) //should be positioned at str.po
+                None
+              }
+              case iuee: IllegalArgumentException => {
```

Comment:
`StringContext.InvalidUnicodeEscapeException` has access to the positioning stuff, but it's not publicly accessible for bincompat reasons. What's the best course of action?

Reply:
> What's the best course of action?

Good question, this is tricky. I guess we could hack the Scala2Unpickler to disregard the private[scala] on this definition but that's pretty ugly.

Reply:
Workarounds with java reflection, going through Java and extending `StringContext$` (or is it final?), parsing the error message (the index is indicated in there) or not refining the position are also possible.

Reply:
> parsing the error message

I suggest doing that for now if it's not too much trouble, and we should leave a TODO for whenever we get to break forward compatibility of scala-library.